src/types.rs

```diff
@@ -238,27 +238,60 @@ pub struct NetworkIdentity {
     pub metadata: MetaData,
 }
 
-#[derive(Serialize, Deserialize)]
+#[derive(Serialize, Deserialize, Debug)]
 pub enum PeerState {
     Disconnected,
     Connecting,
     Connected,
     Disconnecting,
 }
+impl fmt::Display for PeerState {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let printable = match *self {
+            Self::Disconnected => "disconnected",
+            Self::Connecting => "connecting",
+            Self::Connected => "connected",
+            Self::Disconnecting => "disconnecting",
+        };
+        write!(f, "{}", printable)
+    }
+}
 
-#[derive(Serialize, Deserialize)]
+#[derive(Serialize, Deserialize, Debug)]
 pub enum ConnectionOrientation {
     Inbound,
     Outbound,
 }
+impl fmt::Display for ConnectionOrientation {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let printable = match *self {
+            Self::Inbound => "inbound",
+            Self::Outbound => "outbound",
+        };
+        write!(f, "{}", printable)
+    }
+}
 
 #[derive(Serialize, Deserialize)]
 pub struct PeerDescriptor {
     pub state: PeerState,
     pub direction: ConnectionOrientation,
 }
 
-#[derive(Serialize, Deserialize)]
+#[derive(Serialize, Deserialize, Debug)]
+pub struct PeerDescriptionWithMetadata {
+    pub descriptions: Vec<PeerDescription>,
+    #[serde(flatten)]
+    pub meta: PeerMetadata,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct PeerMetadata {
+    #[serde(with = "crate::serde::as_string")]
+    pub count: i32,
+}
```

Comment:
as you have written this now, you are telling the compiler you expect a message that looks like this:

```json
{
  "data": {
    "descriptions": [ { ... a peer description ... }, another, ...  ],
    "count": "some number",
  }
}
```

but if you look at the response schema, e.g. here: https://ethereum.github.io/beacon-APIs/#/Node/getPeers

then you see we want this schema instead:

```json
{
  "data": [ { some peer description }, etc, etc ],
  "meta": { some other stuff },
```

which is handled by the existing `Value` struct, no need to define any additional stuff

Reply:
ok - I started off originally trying to deserialize to `Value<Vec<PeerDescription>>` but that returns the same API error. I'll change it back and see if I can see why.